__tests__/e2e/AddFriendsStateFlow.test.tsx

```diff
@@ -0,0 +1,418 @@
+import React from "react";
+import {
+  render,
+  fireEvent,
+  waitFor,
+  cleanup,
+  act,
+} from "@testing-library/react-native";
+import { NavigationContainer } from "@react-navigation/native";
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import FirestoreCtrl, {
+  DBUser,
+  DBChallenge,
+  DBChallengeDescription,
+  DBGroup,
+  DBComment,
+} from "@/src/models/firebase/FirestoreCtrl";
+import HomeScreen from "@/src/views/home/home_screen";
+import FriendsScreen from "@/src/views/friends/friends_screen";
+import MaximizeScreen from "@/src/views/home/maximize_screen";
+
+const Stack = createNativeStackNavigator();
+
+// Mock FirestoreCtrl
+jest.mock("@/src/models/firebase/FirestoreCtrl", () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      // Mock functions used in home screen
+      getGroupsByUserId: jest.fn((id) => {
+        if (id === "123") {
+          return new Promise<DBGroup[]>((resolve) => {
+            resolve([mockTesterGroup]);
```

Comment:
You can actually use `Promise.resolve()`instead of this more complex version.

Reply:
This way is more complex but it forces the type, which helps mocking the functions